tests/stdlib/topenssl.nim

```diff
@@ -0,0 +1,19 @@
+discard """
+  matrix: "-d:ssl"
+"""
+import openssl
+
+# bug #16308
+block:
+  var rsa: PRSA
+  var file: File
+
+  if not file.open("topenssl_rsa_public_key.pem"):
+    raise newException(Exception, "Error opening file")

```

Comment:
bad error message: error messages should always show runtime context (the filename). Use an auxiliary variable to define the filename to keep it DRY.

Better yet: use  https://github.com/nim-lang/Nim/blob/5514b299ebc1de4ef0c1c182d4f21a02e3c2fa90/lib/system/io.nim#L732 altogether to do this for you